### ISSUE TITLE
az functionapp config appsettings set : "Runtime docker not supported for os linux."

### ISSUE THREAD

**Issue by williamhbell** [Azure/azure-cli#28382]
### Describe the bug

Using:
```
azure-cli                         2.57.0

core                              2.57.0
telemetry                          1.1.0

Dependencies:
msal                              1.26.0
azure-mgmt-resource             23.1.0b2
```

Created an Azure function with:

```
az functionapp create --name "$function_name" `
--storage-account "$storage_account_id" `
--plan "$plan_id" `
--resource-group "$resource_group" `
--functions-version "$functions_version" `
--image "$container_registry.azurecr.io/$image_name" `
--registry-password "$password" `
--registry-username "$username" `
--disable-app-insights
```

This command produces the warning

```
App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.
App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.
```

which refers to [https://github.com/Azure/azure-cli/issues/27724](https://github.com/Azure/azure-cli/issues/27724) .

Once the Azure Function has been created, configuring it with

```
az functionapp config appsettings set `
--name $function_name --resource-group $resource_group `
--settings SOME_SETTING="some value"
```

produces the warning message

```
Runtime docker not supported for os linux. Supported runtimes for os linux are: ['dotnet-isolated', 'dotnet-isolated', 'dotnet-isolated', 'dotnet', 'node', 'node', 'node', 'node', 'python', 'python', 'python', 'python', 'python', 'java', 'java', 'java', 'java', 'powershell', 'custom']. Run 'az functionapp list-runtimes' for more details on supported runtimes.
App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.
```

This appears to be a false warning message, since the container runs as expected on the Linux host.  This warning message has started appearing after updating to a more modern Azure CLI version.

### Related command

az functionapp config appsettings set

### Errors

```
App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.
App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.
```
```
Runtime docker not supported for os linux. Supported runtimes for os linux are: ['dotnet-isolated', 'dotnet-isolated', 'dotnet-isolated', 'dotnet', 'node', 'node', 'node', 'node', 'python', 'python', 'python', 'python', 'python', 'java', 'java', 'java', 'java', 'powershell', 'custom']. Run 'az functionapp list-runtimes' for more details on supported runtimes.
App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.
```

### Issue script & Debug output

This would include too much personal data.

### Expected behavior

Do not expect to see either warning message, since this type of deployment has worked for several years.  The warning messages have only started appearing with a more recent version of the CLI.

### Environment Summary

```
azure-cli                         2.57.0

core                              2.57.0
telemetry                          1.1.0

Dependencies:
msal                              1.26.0
azure-mgmt-resource             23.1.0b2

Python location 'C:\Users\*******\AppData\Local\Programs\Python\Python311\python.exe'
Extensions directory 'C:\Users\*******\.azure\cliextensions'

Python (Windows) 3.11.7 (tags/v3.11.7:fa7a6f2, Dec  4 2023, 19:24:49) [MSC v.1937 64 bit (AMD64)]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by rLindorfer** [comment]
Thanks for referencing https://github.com/Azure/azure-cli/issues/27724, these warning messages still break several build pipelines  in our infrastructure.

**Comment by williamhbell** [comment]
This issue is still present in azure-cli version 2.6.1.  The false error message is shown when a functionapp is configured:
```
Runtime docker not supported for os linux. Supported runtimes for os linux are: ['dotnet-isolated', 'dotnet-isolated', 'dotnet-isolated', 'dotnet', 'node', 'node', 'node', 'node', 'python', 'python', 'python', 'python', 'python', 'java', 'java', 'java', 'java', 'powershell', 'powershell', 'custom']. Run 'az functionapp list-runtimes' for more details on supported runtimes.
App settings have been redacted. Use `az webapp/logicapp/functionapp config appsettings list` to view.
```
The functionapp runs correctly, using Docker on Linux.

**Comment by bslpzk** [comment]
Same issue for `azure-cli` version **2.63.0**

**Comment by jarrodlilkendey** [comment]
Same issue for azure-cli version 2.64.0

> Runtime docker not supported for os linux. Supported runtimes for os linux are: ['dotnet-isolated', 'dotn
> et-isolated', 'dotnet-isolated', 'dotnet', 'dotnet', 'node', 'node', 'python', 'python', 'python', 'python', 'python', 'java', 'java', 'java', 'java', 'powershell', 'powershell', 'custom']. Run 'az functionapp 
> list-runtimes' for more details on supported runtimes.

**Comment by jpr5** [comment]
Same issue for azure-cli 2.67.0.

Why is this issue closed????

**Comment by tuseau** [comment]
I am also experiencing this issue using the latest CLI.

**Comment by crystalworks** [comment]
seeing this in 2.68.0 still.

**Comment by kevin-boutin** [comment]
Please reopen the ticket immediately as this is a known issue with using the azure CLI (specifically the `az functionapp config appsettings set` command).

azure-cli version: 2.74.0

@zhoxing-ms : why did you close this issue without it being resolved?

**Comment by zhoxing-ms** [comment]
Sorry, this issue was previously associated with PR #28867. When PR merges, the issue will also be automatically closed. + @kamperiadis Please note that this issue still has not been resolved yet

**Comment by janny93** [comment]
Observed on latest (v2.82.0) on a `dotnet-isolated` github actions run we attempted today to QA. Had deploy yesterday to dev which worked fine. Re-ran the same github actions deploy to dev today (after seeing this error), and it fails in the same way. Tried to pin to v2.81.0, but got same error.
```
- name: Update appsettings
          uses: azure/CLI@v2
          with:
            azcliversion: latest
            inlineScript: |
              az functionapp config container set \
                -g <resource-group> \
                -n <resource-name> \
                --image <image-name>

              az functionapp config appsettings set \
                -g <resource-group> \
                -n <resource-name> \
                --settings DOCKER_CUSTOM_IMAGE_NAME=<image-name>
```

<img width="1485" height="221" alt="Image" src="https://github.com/user-attachments/assets/eeeedff9-6a9d-4d20-956a-9047e4f0c84f" />

**Update**
One config value A was configured wrongly, which was only seen when an unrelated, and not needed config value B was deleted (with azure cli in commandline as part of troubleshooting). Was resolved by fixing value A. However, strange that az cli error in github action is not able to pinpoint this, but if you have incorrect config values and then delete some other value, only then is the error showing that config value A was not valid

<img width="1682" height="103" alt="Image" src="https://github.com/user-attachments/assets/78cd3518-8c1f-48ff-a75e-3aae2608ab12" />
